ARCHITECTURE.md

```diff
@@ -0,0 +1,252 @@
+This document describes the high-level architecture of `credentialsd`. If you
+want to familiarize yourself with the code base, you are just in the right
+place!
+
+# High-level Overview
+
+There are three APIs defined in [doc/api.md](/doc/api.md). This repository contains
+two services that implement the three APIs defined by the specification:
+
+- `credentialsd`: Implements the Gateway API and Flow Control API
+- `credentialsd-ui`: Implements the UI Control API.
+
+These two services communicate with each other over D-Bus IPC.
+
+The **Gateway** is the entrypoint for clients to interact with. The Flow
+Controler and UI Controller work together to guide the user through the
+process of selecting an appropriate credential based on the request received by
+the Gateway.
+
+The **UI Control API** is used to launch a UI for the user to respond to
+authenticator requests for user interaction. The **Flow Controller** mediates
```

Comment:
Even though we have a dedicated API-doc, I feel like the short explanation of the Flow Controller could be a bit more precise.